src/common/number/numericString.ts

```diff
@@ -98,15 +112,13 @@ export class NumericString {
     // remove the width of negative sign.
     const numLength = coreLength - NumericString.numPrefix[coreRadix].length - (coreSign ? 1 : 0);
 
+    const negative = coreSign;
```

Comment:
This fixes bug with negative flag because coreSign indicates match has `-` prefix,